extensions/liquibase/runtime/src/main/java/io/quarkus/liquibase/runtime/LiquibaseRecorder.java

```diff
@@ -58,10 +58,16 @@ public void doStartActions() {
                             liquibase.dropAll();
                         }
                         if (config.migrateAtStart) {
-                            if (config.validateOnMigrate) {
-                                liquibase.validate();
+                            var lockService = LockServiceFactory.getInstance().getLockService(liquibase.getDatabase());
+                            try {
+                                lockService.waitForLock();
```

Comment:
Is `releaseLock()` "lenient" in that it does _not_ throw an exception if the lock has not been acquired?
What I have in mind is:
- `waitForLock()` fails with an exception
- `finally` kicks in
- `releaseLock()` fails (assumption!) because the lock has not been acquired
- the exception from `finally` shadows the one from `waitForLock()`, confusing the user

I'd move `waitForLock()` out of the `try` block or add a `catch` + `addSuppressed()` to the `finally` block.

This might seem like overengineering, but I had similar cases more often than not.